packages/block-editor/src/hooks/block-bindings.js

```diff
@@ -2,42 +2,116 @@
  * WordPress dependencies
  */
 import { __ } from '@wordpress/i18n';
-import { store as blocksStore } from '@wordpress/blocks';
+import { privateApis as blocksPrivateApis } from '@wordpress/blocks';
 import {
-	BaseControl,
-	PanelBody,
-	__experimentalHStack as HStack,
 	__experimentalItemGroup as ItemGroup,
 	__experimentalItem as Item,
+	__experimentalVStack as VStack,
+	__experimentalText as Text,
+	__experimentalToolsPanel as ToolsPanel,
+	__experimentalToolsPanelItem as ToolsPanelItem,
+	__experimentalTruncate as Truncate,
+	__experimentalDropdownContentWrapper as DropdownContentWrapper,
+	Dropdown,
 } from '@wordpress/components';
-import { useSelect } from '@wordpress/data';
+import { useSelect, useDispatch, useRegistry } from '@wordpress/data';
+import { useContext } from '@wordpress/element';
+import { useViewportMatch } from '@wordpress/compose';
 
 /**
  * Internal dependencies
  */
-import { canBindAttribute } from '../hooks/use-bindings-attributes';
+import {
+	canBindAttribute,
+	getBindableAttributes,
+} from '../hooks/use-bindings-attributes';
+import { store as blockEditorStore } from '../store';
 import { unlock } from '../lock-unlock';
 import InspectorControls from '../components/inspector-controls';
+import BlockContext from '../components/block-context';
+
+const popoverProps = {
+	placement: 'left-start',
+	offset: 36,
+	shift: true,
+};
+
+const useToolsPanelDropdownMenuProps = () => {
+	const isMobile = useViewportMatch( 'medium', '<' );
+	return ! isMobile
+		? {
+				popoverProps: {
+					placement: 'left-start',
+					// For non-mobile, inner sidebar width (248px) - button width (24px) - border (1px) + padding (16px) + spacing (20px)
+					offset: 259,
+				},
+		  }
+		: {};
+};
+
+function BlockBindingsPanelDropdown( {
+	fieldsList,
+	addConnection,
+	attribute,
+} ) {
+	return (
+		<DropdownContentWrapper
+			paddingSize="small"
+			className="block-editor-bindings__popover"
+		>
+			{ Object.entries( fieldsList ).map( ( [ label, fields ] ) => (
+				<ItemGroup
+					key={ label }
+					label={
+						Object.keys( fieldsList ).length > 1 ? label : null
+					}
+				>
+					{ Object.entries( fields ).map( ( [ key, value ] ) => (
+						<Item
+							key={ key }
+							onClick={ () => addConnection( key, attribute ) }
+						>
+							<VStack spacing={ 0 }>
+								<Truncate>{ key }</Truncate>
+								<Truncate className="block-editor-bindings__item-explanation">
+									{ value }
+								</Truncate>
+							</VStack>
+						</Item>
+					) ) }
+				</ItemGroup>
+			) ) }
+		</DropdownContentWrapper>
+	);
+}
+
+function BlockBindingsAttribute( {
```

Comment:
BlockBindingItem maybe? 

I find your option quite verbose. It is true that is each block attribute that can be connected but... 🤔 

It's not a strong opinion, feel free to update it.

Reply:
What about `AttributeBinding` or `BlockAttributeBinding`? In the end, this component takes care of showing the connection of a specific attribute, right?

Reply:
How about `AttributeBinder`? This is the component that actually creates the binding, so I think a name that clearly signals this would be helpful for anyone looking at this file.

Alternative names: `Binder`, `BindingController`.

Reply:
I would say it is not exactly the component that creates the binding. It just shows the attribute and where it is connected to. It is reused in "read-only" mode as well. I would say the one controlling the creation is the Dropdown.